05_iigo/index.tex

```diff
@@ -500,4 +500,226 @@ \subsection{Legislative Branch}
+We have now arrived at the representation of rules we are using for our project. For the rule above:
+\[
+    \textrm{Rule matrix: }
+    \begin{bmatrix}
+        -1 & 1 & -5
+    \end{bmatrix}
+    \textrm{   Variables to fill:  }
+    \begin{bmatrix}
+        \textrm{Amount of tax agent is asked to pay} \\
+        \textrm{Amount of tax the agent is actually paying} \\
+        1
+    \end{bmatrix}     
+\]
+\[
+    \textrm{   Auxiliary vector: } 
+    \begin{bmatrix}
+        2 
+    \end{bmatrix}   
+\]
+Let us revist our tax rule once more, adding a further layer of complexity.
+\linebreak
+\linebreak
+"You must pay in tax at least half as many resources as you have and if you've been told to pay tax x you must pay at least tax of x."
+In this example there are two separate conditions. We can write both conditions as separate rules.
+Condition 1:
+\[
+    \textrm{Rule matrix: }
+    \begin{bmatrix}
+        0 & 2 & -1 & 0
+    \end{bmatrix}
+    \textrm{   Variables to fill:  }
+    \begin{bmatrix}
+        \textrm{Amount of tax agent is asked to pay} \\
+        \textrm{Amount of tax the agent is actually paying} \\
+        \textrm{Amount of resources agent has} \\
+        1
+    \end{bmatrix}     
+\]
+\[
+    \textrm{   Auxiliary vector: } 
+    \begin{bmatrix}
+        2 
+    \end{bmatrix}   
+\]
+\linebreak
+Condition 2:
+\[
+    \textrm{Rule matrix: }
+    \begin{bmatrix}
+        -1 & 1 & 0 & 0
+    \end{bmatrix}
+    \textrm{   Variables to fill:  }
+    \begin{bmatrix}
+        \textrm{Amount of tax agent is asked to pay} \\
+        \textrm{Amount of tax the agent is actually paying} \\
+        \textrm{Amount of resources agent has} \\
+        1
+    \end{bmatrix}     
+\]
+\[
+    \textrm{   Auxiliary vector: } 
+    \begin{bmatrix}
+        2 
+    \end{bmatrix}   
+\]
+However, we can also write this as a single stacked rule:
+\[
+    \textrm{Rule matrix: }
+    \begin{bmatrix}
+        0 & 2 & -1 & 0 \\
+        -1 & 1 & 0 & 0 
+    \end{bmatrix}
+    \textrm{   Variables to fill:  }
+    \begin{bmatrix}
+        \textrm{Amount of tax agent is asked to pay} \\
+        \textrm{Amount of tax the agent is actually paying} \\
+        \textrm{Amount of resources agent has} \\
+        1
+    \end{bmatrix}     
+\]
+\[
+    \textrm{   Auxiliary vector: } 
+    \begin{bmatrix}
+        2 \\
+        2 
+    \end{bmatrix}   
+\]
+The real power of this representation of rules is that the mechanics of the calculation is exposed. 
+Any agent can inspect the elements of the rule matrix and simply calculate what values they must provide to pass the rule.
+For example, if an agent is looking at the above rule and knows that their resource count is 120, and also know they've been asked to pay 75, they can go line by line through the matrix to work our what the minimum amount of tax that is needed to be paid.
```

Comment:
above.. hmm....

maybe, we should define some environment for rules..
can you try 
`\begin{rule} \end{rule}` plz !!!
and THEN we can reference them easilyyyyy